cmd/kueuectl/app/util/dry_run.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+type DryRunStrategy int
+
+const (
+	// DryRunNone indicates the client will make all mutating calls
+	DryRunNone DryRunStrategy = iota
+
+	// DryRunClient, or client-side dry-run, indicates the client will prevent
+	// making mutating calls such as CREATE, PATCH, and DELETE
+	DryRunClient
+
+	// DryRunServer, or server-side dry-run, indicates the client will send
+	// mutating calls to the APIServer with the dry-run parameter to prevent
+	// persisting changes.
+	//
+	// Note that clients sending server-side dry-run calls should verify that
+	// the APIServer and the resource supports server-side dry-run, and otherwise
+	// clients should fail early.
+	//
+	// If a client sends a server-side dry-run call to an APIServer that doesn't
+	// support server-side dry-run, then the APIServer will persist changes inadvertently.
+	DryRunServer
+)
+
+// AddDryRunFlag adds dry-run flag to a command.
+func AddDryRunFlag(cmd *cobra.Command) {
+	cmd.Flags().String(
+		"dry-run",
+		"none",
+		`Must be "none", "server", or "client". If client strategy, only print the object that would be sent, without sending it. If server strategy, submit server-side request without persisting the resource.`,
```

Comment:
It is a bit ambiguous. Does it print yaml if server mode is used?

Reply:
No. It's just return result success or not.

Reply:
```shell 
$ kubectl kueue create lq lq1 -c cq1 --dry-run=server              
> localqueue.kueue.x-k8s.io/lq1 created (server dry run)
```

Reply:
Oh, yes, if we put -o yaml

Reply:
```shell
$ kubectl kueue create lq lq1 -c cq1 --dry-run=server -o yaml

apiVersion: kueue.x-k8s.io/v1beta1
kind: LocalQueue
metadata:
  creationTimestamp: "2024-04-30T20:57:06Z"
  generation: 1
  name: lq1
  namespace: default
  uid: f4683c4c-e074-41b4-8ab1-75b3e8b21519
spec:
  clusterQueue: cq1
status:
  admittedWorkloads: 0
  flavorUsage: null
  flavorsReservation: null
  pendingWorkloads: 0
  reservingWorkloads: 0

```

Reply:
Interesting, they did a lot of work in the framework to build kubectl plugins. That's awesome.